### ISSUE TITLE
IRclNode Context property is concrete

### ISSUE THREAD

**Issue by stewpend0us** [noelex/rclnet#16]
I was setting up a "Fake" Rcl system so that I can at least launch my GUI without having ROS installed. I got almost all the way to the end before I realized that the "Context" property on my "FakeNode" is being called by the Rcl library itself. But since I can't construct a RclContext without having ROS installed I'm right back where I started.

I'm sure there's a good reason why not...but can the type of the Context property be IRclContext instead of RclContext?

**Comment by noelex** [comment]
Yeah that was my intention, as you can see from the XML documentation. I’ll fix this.

**Comment by stewpend0us** [comment]
Ah. I'm not building...Also having the same problem with RclClock (which you also fixed) any idea when this might be released to NuGet?

**Comment by stewpend0us** [comment]
another one:

```IRclTimer CreateTimer(RclClock clock, Ti...```

needs to be IRclClock

**Comment by noelex** [comment]
This way already fixed on devel branch. Maybe I can publish a preview package tomorrow.

**Comment by stewpend0us** [comment]
not sure if this one is really a problem or not

https://github.com/noelex/rclnet/blob/d81d7996321f51644bee281d30287de2362f08cc/src/Rcl.NET/IRclContext.cs#L81

**Comment by noelex** [comment]
Nope, you can create `YieldAwaiter` without `SynchronizationContext` and this won’t have any side-effect.

**Comment by stewpend0us** [comment]
surely this one is though?
https://github.com/noelex/rclnet/blob/d81d7996321f51644bee281d30287de2362f08cc/src/Rcl.NET/IRclNode.cs#L42

**Comment by noelex** [comment]
Good catch. This needs to be an interface.

**Comment by stewpend0us** [comment]
Hey,
Not sure where/how to ask you so I'll just do it here...
What's the relationship between ros2cs and rcl.net? I noticed your most recent commits all seem to be related to? or patching? ros2cs? I've submitted issues on that repo but never gotten a reply....I assumed when you 'install' ros2cs it was coming from the other repo?

Thanks!

**Comment by noelex** [comment]
No, there’s no relationship between the ros2cs and this repository.

‘ros2cs’ in this repository is a code generation tool for generating C# code for ROS interface definitions, which just happens to be the same name with the repository you mentioned.

**Comment by stewpend0us** [comment]
> No, there’s no relationship between the ros2cs and this repository.
> 
> ‘ros2cs’ in this repository is a code generation tool for generating C# code for ROS interface definitions, which just happens to be the same name with the repository you mentioned.

Ok. So when I install ros2cs from nuget it is coming from this repo. Only took me 6 months to figure that out...ugh! I'll move the issues over here. I think you already started working on them though
